Objects/longobject.c

```diff
@@ -5260,6 +5260,36 @@ long_is_finite(PyObject *v)
 }
 #endif
 
+/*[clinic input]
+int.as_integer_ratio
+
+Return integer ratio.
+
+Return a pair of integers, whose ratio is exactly equal to the original int
+and with a positive denominator.
+
+>>> (10).as_integer_ratio()
+(10, 1)
+>>> (-10).as_integer_ratio()
+(-10, 1)
+>>> (0).as_integer_ratio()
+(0, 1)
+[clinic start generated code]*/
+
+static PyObject *
+int_as_integer_ratio_impl(PyObject *self)
+/*[clinic end generated code: output=e60803ae1cc8621a input=55ce3058e15de393]*/
+{
+    if PyLong_CheckExact(self) {
+        return PyTuple_Pack(2, self, _PyLong_One);
+    } else {
+        PyObject *numerator = _PyLong_Copy(self);
+        PyObject *ratio_tuple = PyTuple_Pack(2, numerator, _PyLong_One);
+        Py_DECREF(numerator);
+        return ratio_tuple;
+    }
```

Comment:
Could also write the following to avoid repetition:
```C
PyObject *numerator;
if (PyLong_CheckExact(self)) {
    numerator = self;
    Py_INCREF(numerator);
}
else {
    numerator = _PyLong_Copy(self);
    if (numerator == NULL) {
        return NULL;
    }
    PyObject *ratio_tuple = PyTuple_Pack(2, numerator, _PyLong_One);
    Py_DECREF(numerator);
    return ratio_tuple;
}

Reply:
ISTM, the code is reasonable as-is.